app/js/actions/jobs.js

```diff
@@ -14,6 +14,40 @@ export const jobCompleted = (job) => ({
     job
 });
 
+export const registerPath = (visualization) => ({
+    type: 'REGISTER_PATH',
+    visualization
+});
+
+export const watchForVisualization = (jobUUID, router, url) => {
+    return (dispatch, getState) => {
+        const { connection: { secretKey, uri } } = getState();
+        const update = () => {
+            const job = getState().jobs.completedJobs.find(j => j.uuid === jobUUID);
+            if (job) {
+                const visUUID = job.outputs.visualization;
+                let vis = getState().artifacts.visualizations.find(v => v.uuid === visUUID);
+                if (!vis) {
+                    setTimeout(update, 1000);
+                    return;
+                }
+                fetchAPI(secretKey, 'GET', `http://${uri}/api/workspace/view/${vis.uuid}`)
+                .then((json) => {
+                    vis = {
+                        ...vis,
+                        filePath: json.filePath
+                    };
+                    return dispatch(actions.registerPath(vis));
+                })
+                .then((action) => router.push(`${url}/${action.visualization.uuid}`));
+            } else {
+                setTimeout(update, 1000);
```

Comment:
It works really well too, I was able to mash `go` (which we need to fix) and each timeout independently ended when the job finished.